File: backend/components/metiers/administration_publique/utils/validators/samples/sample_usage.test.py
# sample_usage.test.py
"""Tests unitaires avancés pour les exemples validators Python"""
from ..core import validators
import json
import pytest

def test_validate_email():
    with open('sample_validators_data.json') as f:
        data = json.load(f)
    assert validators.validate_email(data['email'])

def test_validate_required():
    with open('sample_validators_data.json') as f:
        data = json.load(f)
    assert validators.validate_required(data['requiredField'])
    assert not validators.validate_required('')

File: backend/components/metiers/administration_publique/utils/validators/samples/sample_usage.test.js
// sample_usage.test.js
// Tests unitaires avancés pour les exemples validators JS
const validators = require('../core/validators');
const data = require('./sample_validators_data.json');

describe('Sample usage validators JS', () => {
  it('valide un email', () => {
    expect(validators.validateEmail(data.email)).toBe(true);
  });
  it('valide un champ obligatoire', () => {
    expect(validators.validateRequired(data.requiredField)).toBe(true);
  });
  it('échoue sur un champ vide', () => {
    expect(validators.validateRequired('')).toBe(false);
  });
});
